zcm/julia/ZCM.jl

```diff
@@ -0,0 +1,334 @@
+module ZCM
+
+# RRR / Note: Julia requires that the memory layout of the C structs is consistent
+#             between their definitions in zcm headers and this file
+
+# C ptr types for types that we don't need the internals of
+module Native
+
+type Zcm
```

Comment:
I think a lot of these `type`s could/should be `immutable`s (though I'm not 100% sure about what's required by the `ccall` interface; @rdeits has more experience with that).

Also, in Julia 0.6 (and future 1.0), `type` will be `mutable struct` and `immutable` will be `struct`.

Reply:
Anything that has a finalizer has to be immutable, otherwise yes I agree.

Reply:
Errr, nope, exactly wrong. Anything with a finalizer has to be *mutable*